plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogConfig.java

```diff
@@ -30,4 +33,18 @@ public IcebergGlueCatalogConfig setCacheTableMetadata(boolean cacheTableMetadata
         this.cacheTableMetadata = cacheTableMetadata;
         return this;
     }
+
+    @Min(1)
+    public int getMetadataParallelism()
+    {
+        return metadataParallelism;
+    }
+
+    @ConfigDescription("Limits metadata enumeration calls parallelism")
+    @Config("iceberg.glue.metadata.parallelism")
```

Comment:
I don't think we need to make this configuration per catalog type. One configuration in IcebergConfig should be fine, each implementation can choose to use this or not.
The only way it would make sense to have separate configs is if we find that there is a big difference in the amount of parallelism we can squeeze from the different metastores.